### ISSUE TITLE
SetRenderState(D3DRS_DEPTHBIAS, g_depth_bias) causing depth buffer issues

### ISSUE THREAD

**Issue by nukeulater** [PermaNulled/cartographer#118]
![capture](https://user-images.githubusercontent.com/18323679/38452706-063eb1ac-3a52-11e8-8a87-cdce0ed016b2.PNG)

![unknown1](https://user-images.githubusercontent.com/18323679/38452705-0131d770-3a52-11e8-9a19-eef6732c8fa3.png)

![unknown](https://user-images.githubusercontent.com/18323679/38452704-fb557d20-3a51-11e8-8d0d-72281f1ee509.png)

If you look closely you can see lights / transparent planes shine through geometry.
Update 2022:
~~Happens on MCC too.~~ 
H2MCC patched in 2021 by 343i 👍🏻 ([link](https://support.halowaypoint.com/hc/en-us/articles/4407245081620-Halo-The-Master-Chief-Collection-MCC-Season-8-Patch-Notes-October-2021-))

> Lights emitted by enemies in Campaign no longer show through some solid objects briefly when the enemy moves out of view.

They start to behave as expected if you get closer to the light / transparent plane source.

**Comment by korpser32** [comment]
Yes this happens on my PC as well as on my Xbone with MCC on classic Halo 2 mode. I believe this is a result of the modded engine to allow 60 fps, as it was originally intended and all physics coded to be tied to the framerate. The same thing happens on PC again because the game has been modified to allow a higher than planned framerate and could mess up the calculations for when to draw lighting on your screen, because as you mentioned when you get closer it gets fixed and you can no longer see the light through the wall. I wouldn't worry too much about it, I get it's immersion breaking because I've noticed and felt that way too but I just take it as the one little downside to a great game.

**Comment by nukeulater** [comment]
The "tickrate" change has nothing to do with this. I personally tested that by setting the game engine back to xbox values (possible on PC) and the issue still occurred.
Most likely a ZBuffer issue.

**Comment by General-101** [comment]
Depth bias offset might be the cause or somehow related to this. Setting it to zero hides the issue.

**Comment by nukeulater** [comment]
Fixed if 62e8d80f2b4057c5bfd6c40fddb9b0964be7f114 is merged.

**Comment by iSupersniper** [comment]
This change causes zflighting issues with all bitmaps that are modified when looked at from a distance.

![capture 1](https://user-images.githubusercontent.com/17890753/44311394-912ebc00-a3b4-11e8-841d-3da6d98db4cd.png)

**Comment by nukeulater** [comment]
Z-figthing happens because that light is coplanar with the wall behind/it has similar or identical values inside the Z-buffer w/ the wall.
Unless you can find a way to move that light a bit or determine from which shader the depth bias value is taken from (and I suspect most of the plane lights use the same shader), it cannot be fixed.

**Comment by General-101** [comment]
Just to be clear for anyone reading. The zfighting only happens if the shader used for that particular mesh had a depth bias offset value defined. This basically means that the only zfighting you should see is on mesh pieces that showed through geometry in the first place.
